presto-common/src/main/java/com/facebook/presto/common/type/ArrayType.java

```diff
@@ -86,10 +86,14 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
         }
 
         for (int i = 0; i < leftArray.getPositionCount(); i++) {
-            checkElementNotNull(leftArray.isNull(i), ARRAY_NULL_ELEMENT_MSG);
-            checkElementNotNull(rightArray.isNull(i), ARRAY_NULL_ELEMENT_MSG);
-            if (!elementType.equalTo(leftArray, i, rightArray, i)) {
-                return false;
+            if (leftArray.isNull(i) == rightArray.isNull(i)) {
+                if (!elementType.equalTo(leftArray, i, rightArray, i)) {
+                    return false;
+                }
+            }
+            else {
+                checkElementNotNull(leftArray.isNull(i), ARRAY_NULL_ELEMENT_MSG);
```

Comment:
this is not right. we intentionally throw for equals comparison with nulls for arrays and maps (and if we did change it, we would want to return false rather than throw).